indico/modules/events/persons/controllers.py

```diff
@@ -148,7 +148,9 @@ def get_persons(self):
 
             if event_person.user:
                 for role in event_person.user.event_roles:
```

Comment:
`.filter_by(event=self.event)`

Reply:
The relationship is not dynamic.
`AttributeError: 'InstrumentedSet' object has no attribute 'filter_by'`

Reply:
OK, would be causing query spam anyway since it's in a loop..

But...o this always loads all event roles of the user - and since we don't subqueryload/joinedload it, it's an extra query for every single user anyway. Maybe we can build a dict of user->roles mappings only for people in the event outside the loop?